src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAGenerateRuntimeConfigurationFiles.cs

```diff
@@ -0,0 +1,248 @@
+                        {
+                            {"FrameworkName", "Microsoft.AspNetCore.App"}, {"Version", "3.1.0"}
+                        }
+                    )
+                },
+                RollForward = "LatestMinor"
+            };
+
+            Action a = () => task.PublicExecuteCore();
+            a.ShouldNotThrow();
+
+            File.ReadAllText(_runtimeConfigPath).Should()
+                .Be(
+                    @"{
+  ""runtimeOptions"": {
+    ""tfm"": ""netcoreapp3.0"",
+    ""rollForward"": ""LatestMinor"",
+    ""frameworks"": [
+      {
+        ""name"": ""Microsoft.WindowsDesktop.App"",
+        ""version"": ""3.1.0""
+      },
+      {
+        ""name"": ""Microsoft.AspNetCore.App"",
+        ""version"": ""3.1.0""
+      }
+    ]
+  }
+}",
+                    "There is no Microsoft.NETCore.App. And it is under frameworkS.");
+        }
+
+        [Fact]
+        public void Given2RuntimeFrameworksItCanGenerateWithoutAssetFile()
+        {
+            var task = new TestableGenerateRuntimeConfigurationFiles
+            {
+                BuildEngine = new MockBuildEngine4(),
+                TargetFrameworkMoniker = ".NETCoreApp,Version=v3.0",
+                TargetFramework = "netcoreapp3.0",
+                RuntimeConfigPath = _runtimeConfigPath,
+                RuntimeConfigDevPath = _runtimeConfigDevPath,
+                RuntimeFrameworks = new[]
+                {
+                    new MockTaskItem(
+                        "Microsoft.NETCore.App",
+                        new Dictionary<string, string>
+                        {
+                            {"FrameworkName", "Microsoft.NETCore.App"}, {"Version", "3.1.0"}
+                        }
+                    ),
+                    new MockTaskItem(
+                        "Microsoft.WindowsDesktop.App",
+                        new Dictionary<string, string>
+                        {
+                            {"FrameworkName", "Microsoft.WindowsDesktop.App"}, {"Version", "3.1.0"}
+                        }
+                    )
+                },
+                RollForward = "LatestMinor"
+            };
+
+            Action a = () => task.PublicExecuteCore();
+            a.ShouldNotThrow();
+
+            File.ReadAllText(_runtimeConfigPath).Should()
+                .Be(
+                    @"{
+  ""runtimeOptions"": {
+    ""tfm"": ""netcoreapp3.0"",
+    ""rollForward"": ""LatestMinor"",
+    ""framework"": {
+      ""name"": ""Microsoft.WindowsDesktop.App"",
+      ""version"": ""3.1.0""
+    }
+  }
+}",
+                    "There is no Microsoft.NETCore.App.");
+        }
+
+
+        private class TestableGenerateRuntimeConfigurationFiles : GenerateRuntimeConfigurationFiles
+        {
+            public void PublicExecuteCore()
+            {
+                base.ExecuteCore();
+            }
+        }
+
+        private class MockBuildEngine4 : MockBuildEngine, IBuildEngine4
```

Comment:
Do we not have one of these somewhere that can be reused?

Reply:
That's the only place we use MockBuildEngine**4** since we have more logic about the cache has dependency on it